Sources/Apodini/Modifier/DelegationModifier.swift

```diff
@@ -88,47 +103,53 @@ public extension DelegatingHandlerInitializer {
 
 // MARK: DelegatingHandlerContextKey
 
-struct DelegatingHandlerContextKey: ContextKey {
-    typealias Value = [(Bool, AnyDelegatingHandlerInitializer)]
-    static var defaultValue: Value = []
-}
-
+public struct DelegatingHandlerContextKey: ContextKey {
+    public typealias Value = OrderedSet<Entry>
+    public static var defaultValue: Value = []
 
-// MARK: DelegatingHandlerInitializerVisitor
+    public static func reduce(value: inout Value, nextValue: Value) {
+        // append won't update members already in the original set (see `Entry.ensureInitializerTypeUniqueness`)
+        value.append(contentsOf: nextValue)
+    }
+}
 
-class DelegatingHandlerInitializerVisitor: HandlerVisitor {
-    var initializers: [AnyDelegatingHandlerInitializer]
-    let semanticModelBuilder: SemanticModelBuilder?
-    unowned let visitor: SyntaxTreeVisitor
-    
-    init(calling builder: SemanticModelBuilder?, with visitor: SyntaxTreeVisitor) {
-        let initializers = visitor.currentNode.peekValue(for: DelegatingHandlerContextKey.self)
-        self.initializers = (initializers.filter { prepend, _ in prepend }.reversed()
-                                + initializers.filter { prepend, _ in !prepend }).map { _, initializer in initializer }
-        self.semanticModelBuilder = builder
-        self.visitor = visitor
+extension DelegatingHandlerContextKey {
+    /// Represents the entry type for the value of an ``DelegatingHandlerContextKey``
+    public struct Entry {
+        /// Every entry of the ``DelegatingHandlerContextKey`` is identified by a instance specific
+        /// `UUID` used to check if we already inserted into the ``Handler`` stack when parsing initializers.
+        private let uuid: UUID
+
+        /// The according ``AnyDelegatingHandlerInitializer`` used to instantiate the delegating ``Handler``.
+        let initializer: AnyDelegatingHandlerInitializer
+        /// If set to true, it is ensure that the same ``AnyDelegatingHandlerInitializer``, even though
+        /// when inserted multiple times into the context, is only used a single time (the first time it got inserted).
```

Comment:
What happens when you insert once, then the insertion gets filtered, and then you insert a second time?

Reply:
Depending on your level of confidence on this edge case, you might want to add a test-case😉

Reply:
> What happens when you insert once, then the insertion gets filtered, and then you insert a second time?

The enforcement of the `ensureInitializerTypeUniqueness` is done through the `OrderedSet` (see the `Hashable` implementation of the ContextKey value). So a second value won't ever be added to the set (Remember, while we build the set incrementally in the Visitor, the context key value in the `ContextNode` will always contain all values ever added). Therefore the filter can always just apply to the "first" instance.

But good thought nonetheless. Agree it might be good to add a test case for that, just to make sure no regression introduces the described bug when deciding to e.g. exchanging the OrderedSet implementation.

Reply:
According to our discussion, with the latest commit, I adjusted the behavior, such that a unique Initializer is added nonetheless, if there was the same initializer, but filtered by a according DelegationFilter.